### ISSUE TITLE
Better comment format

### ISSUE THREAD

**Issue by bobbinth** [maticnetwork/miden#82]
Currently, comments in Miden assembly must be surround by `#` characters. Thus, even for single line comments we must write:

```
# this is a comment #
```

It is a bit annoying that we need to terminate single line comments with `#`. The main motivation for doing so was ease of parsing, and in fact parsing this style of comments is very easy (they are stripped out from the source code as implemented [here](https://github.com/maticnetwork/miden/blob/next/assembly/src/tokens/stream.rs#L25). But maybe there are fairly simple ways to strip out comments which terminate with a newline instead of `#` as well?

**Comment by itzmeanjan** [comment]
Yes I've to agree with you, as per my primary experience of writing Miden Assembly comments. What do you think of using similar syntax to support multi-line comments ?

So one writes

```
# single line comment #
proc.func
end

begin
exec.func
end
```

also may write

```
# multi
line 
comment #
proc.func
end

begin
exec.func
end
```

Because I think, we have to support multi-line comments too.

**Comment by bobbinth** [comment]
Yeah - I think multi-line comments would be useful too. I guess we have the following potential styles:

- single-line comment terminated withe a newline character
- multi-line comment

We also want to make sure we preserve easy parsing, and with this, i don't think we can use the same character for both. So, maybe we keep the current approach for multi-line comments and come up with a new character for single-line comments?

So, for example, this should already work:

```
# multi
line 
comment #
proc.func
end

begin
exec.func
end
```

And maybe we can add something like this:
```
begin
exec.func   ! single line comment
end
```

Also, open to using other characters for either style.

**Comment by grjte** [comment]
> It is a bit annoying that we need to terminate single line comments with #.
I'm also not a fan of the terminating character requirement for single-line comments. It should not be hard to strip out comments terminating in newline - I'm pretty sure I've done this in a previous life :)

> I guess we have the following potential styles:
>
> * single-line comment terminated withe a newline character
> * multi-line comment
I agree that providing block comments as an option generally makes things easier for development.

> We also want to make sure we preserve easy parsing, and with this, i don't think we can use the same character for both
Agreed. However, if we do this then my preference would be to use a set of line and block comments that's already common, such as the classic pair of ```// comment``` and ```/* multi-line comment */``` from C/Javascript/Rust, etc

I think this reduces cognitive load for the user as opposed to defining new comment styles.

If sticking with a single line comments only, then continuing with the ```#``` like in python makes sense (simple parsing)

**Comment by bobbinth** [comment]
Thinking more about, I'm a bit split on block comments. For example, in my own usage, the only times I'm using true block comments is when I want to comment out big pieces of code. This is useful, but I think with modern tools this can be just as easily accomplished with multi-line editing. So, I guess the two options are:

### Option 1
Keep the `#` is the character for comments, but change the semantics to terminate comments at line's end.

```
# Kind of multi-
# line comment
begin
exec.func      # single line comment
end
```

### Option 2
Adopt C-style comments (I agree that it's better than inventing our own style).

```
/*
Multi-
line comment
*/
begin
exec.func      // single line comment
end

**Comment by grjte** [comment]
I would say let's do Option 1 for now because it's straightforward. As you said - with modern tools, it's not as big a deal, so we could table Option 2 and see during ```stdlib``` development how much of a hindrance the lack of block comments actually turns out to be in practice

**Comment by grjte** [comment]
Closed by #164
